### PR TITLE
refactor: add edit modal tests

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -**
Adds docstrings and tests for EditModal and child components.

JIRA: [AU-3](https://openedx.atlassian.net/browse/AU-3)
(Partial completion of ticket)

**What changed?**

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [x] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: src/components/GradesTab/EditModal/OverrideTable/index.jsx [hunk 1]
```diff
@@ -16,6 +16,11 @@ const GRADE_OVERRIDE_HISTORY_COLUMNS = [
   { label: 'Adjusted grade', key: 'adjustedGrade' },
 ];
 
+/**
+ * <OverrideTable />
+ * Table containing previous grade override entries, and an "edit" row
+ * with todays date, an AdjustedGradeInput and a ReasonInput
+ */
```
Comment (nsprenkle) [inline src/components/GradesTab/EditModal/OverrideTable/index.jsx:23]: Not part of the PR but it seems _really weird_ the way this table is set up and I want to whine about it: The `<AdjustedGradeInput/>`/`<ReasonInput/>` feel like this should just be replaced with a `<AdjustGradeControls/>` which makes up the final row of the table. Breaking out the extra controls seems too piecemeal.

File: src/components/GradesTab/EditModal/index.jsx [hunk 2]
```diff
@@ -16,6 +16,15 @@ import thunkActions from 'data/thunkActions';
 import OverrideTable from './OverrideTable';
 import ModalHeaders from './ModalHeaders';
 
+/**
```
Comment (nsprenkle) [inline src/components/GradesTab/EditModal/index.jsx:19]: Thank you for all these beautiful docstrings 🎉

File: src/components/GradesTab/EditModal/HistoryHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * HistoryHeader
+ * simple display container for an individual history table header
+ * @param {string} id - header id
+ * @param {string} label - header label
+ * @param {string} value - header value
+ */
+const HistoryHeader = ({ id, label, value }) => (
+  <div>
```
Comment (nsprenkle) [inline src/components/GradesTab/EditModal/HistoryHeader.jsx:12]: Proposed refactor: make this a `<table>` or `grid` instead of `<div>`s masquerading poorly as a table.

**Review by nsprenkle — APPROVED** [review]
Visual and functional review look good. Proposed some style/refactoring around areas this code modifies.

File: src/components/GradesTab/EditModal/OverrideTable/index.jsx (hunk 1 above)
Comment (muselesscreator) [reply src/components/GradesTab/EditModal/OverrideTable/index.jsx:23]: also: objects (or object-creator functions that aren't react components) can't be "connect-ed" to redux, which is kind of frustrating.

File: src/components/GradesTab/EditModal/HistoryHeader.jsx (hunk 3 above)
Comment (muselesscreator) [reply src/components/GradesTab/EditModal/HistoryHeader.jsx:12]: I would put that as a new ticket.  We also have to updgrade both our tables to be using DataTable anyway, so makes sense to put some thought into how we wanna do that.

**Review by leangseu-edx — APPROVED** [review]
